assets/tests/consolidated_1_1_tests.yaml

```diff
@@ -5212,12 +5212,18 @@ tests:
           - user: document:1#allowed
             relation: viewer
             object: document:1
+        checkAssertions:
+          - tuple:
+              user: user:jon
+              relation: viewer
+              object: document:1
+            expectation: false
         listObjectsAssertions:
           - request:
               user: user:jon
               type: document
               relation: viewer
-            errorCode: 2002
+            expectation: []
```

Comment:
These test changes are the 1 thing I'm not too sure about. I'd like to discuss this with the @openfga/backend team more.

If you have the following model and tuples:

```
type user

type document
  relations
    define allowed: [user, document#viewer]
    define viewer: [user, document#editor] and allowed

- document:1#viewer@user:jon
- document:1#viewer@document:1#editor
- document:1#editor@document:1#viewer
```
Before this PR `Check(document:1#viewer@user:jon)` would return an ErrCycleDetected. With this PR we return `{allowed: false}, nil` instead. The idea being that if you could have resolved a branch you would have, but otherwise you looped back around and thus didn't find a resolution path.

Reply:
Just to be abundantly clear as well, this is a breaking change of behavior. If we agree it makes sense to proceed forward with it that is ok, but I wanted to at least point it out.

Reply:
**Note** The above model is missing the `editor` relationship.

But otherwise, I softly disagree about this being a breaking change because the results are now completely deterministic and not relying on race conditions. So I see this more as a bug fix 😄 .

Reply:
I'm pro this. 

But we should start considering a way to produce events so folks can debug/improve their models accordingly. Unsure if logging would cut it.

Reply:
> Unsure if logging would cut it.

`if err == cycle { span.setError(cycle) }`

please :)

Reply:
@miparnisari now we also record an error if we see a branch with a cycle (or error in general).

<img width="1236" alt="Screenshot 2024-02-16 at 2 55 46 PM" src="https://github.com/openfga/openfga/assets/2899204/29a38925-b983-413e-a86a-a79e522a070a">